CONTRIBUTING.md

```diff
@@ -0,0 +1,214 @@
+# Contribute to the secrets of sonar-text
+
+## Development environment
+
+1. Set up a Java development environment, or use Docker, to be ready to perform commands such as those specified in the [README.md](README.md)
+2. Set up an extension that validates YAML files, such as [YAML by RedHat](https://github.com/redhat-developer/vscode-yaml)
+3. Set up this [json-schema file](https://github.com/SonarSource/sonar-text/blob/master/sonar-text-plugin/src/main/resources/org/sonar/plugins/secrets/configuration/specifications/specification-json-schema.json) for your validation tooling
+4. Take a look at [the existing rules](https://github.com/SonarSource/sonar-text/tree/master/sonar-text-plugin/src/main/resources/org/sonar/plugins/secrets/configuration)
+5. Familiarize yourself with [SonarSource/rspec](https://github.com/SonarSource/rspec)
+
+## Adding a secret
+
+1. Write the yaml file
+2. Validates the yaml against the schema file
+3. use the script `secretSpecificationInclusionGenerator.sh <Rule-Api jar>`
+  1. It will generate the java files that will glue the yaml file to the code base
+  2. If the Rspec is already available, it will also generate additional rspec data
+4. Test the code:
+  1. If rspec data is available, do `mvn clean test`
+  2. Else, do `mvn test -Dtest=FakecloudCheckTest` and replace this test name by the one generated previously 
+
```

Comment:
The next steps are missing:
5. Merge the RSPEC PR once approved
6. At this time, generates the RSPEC HTML files:
```
pushd sonarpedia-secrets
java -jar ../rule-api-2.7.0.2609.jar generate -rule SXXXX 
```